doc/frame-messenger.md

```diff
@@ -38,6 +41,65 @@ The `open` function can `return` an optional cleanup function, which is called w
 
 ## axe.frameMessenger({ post })
 
-`post` is a function that dictates how axe-core communicates with frames. It is passed three arguments: `frameWindow`, which is the frames `contentWindow`, the `data` object, and a `replyHandler` that must be called when responses are received.
+`post` is a function that dictates how axe-core communicates with frames. It is passed three arguments: `frameWindow`, which is the frames `contentWindow`, the `data` object, and a `replyHandler` that must be called when responses are received. To inform axe-core that no message was sent, return `false`. This informs axe-core not to await for the ping to time out.
+
+Currently, axe-core will only require `replyHandler` to be called once, so promises can also be used here. This may change in the future, so it is preferable to make it possible for `replyHandler` to be called multiple times. Some axe-core [plugins](plugins.md) may rely on this feature.
+
+A second frameMessenger feature available to plugins, but not used in axe-core by default is to reply to a reply. This works by passing `replyHandler` a `responder` callback as a second argument. This requires a different setup, in which callbacks as stored based on their `channelId` property.
+
+```js
+// store handlers based on channelId
+const channels = {};
+
+axe.frameMessenger({
+  post(frameWindow, data, replyHandler) {
+    // Store the handler so it can be called later
+    channels[data.channelId] = replyHandler;
+    // Send a message to the frame
+    bridge.send(frameWindow, data);
+  },
+
+  open(topicHandler) {
+    function subscriber(frameWin, data, response) {
+      const { channelId, message, messageId } = data;
+      // Create a callback to invoke on a reply.
+      const responder = createResponder(frameWin, channelId);
+
+      // If there is a topic, pass it to the axe supplied topic-handler
+      if (data.topic) {
+        topicHandler(data, responder);
+
+        // If there is a replyHandler stored, invoke it
+      } else if (channels[channelId]) {
+        const replyHandler = channels[channelId];
+        replyHandler(message, keepalive, responder);
+
+        // Clean up replyHandler, as no further messages are expected
+        if (!keepalive) delete channels[channelId];
+      }
+    }
+
+    // Start listening for "axe-core" events
+    const unsubscribe = bridge.subscribe('axe-core', subscriber);
+    // Tell axe how to close the connection if it needs to
+    return unsubscribe;
+  }
+});
+
+// Return a function to be called when a reply is received
+function createResponder(frameWin, channelId) {
+  return function responder(message, keepalive, replyHandler) {
+    // Store the new reply handler, possibly replacing a previous one
+    //   to avoid receiving a message twice.
+    channels[channelId] = replyHandler;
+    // Send a message to the frame
+    bridge.send(frameWin, { channelId, message, keepalive });
+  };
+}
+```
+
+## Error handling & Timeouts
```

Comment:
The typings for `Responder` got updated to suggest that it needs to be able to accept a message of type `Error`, but this section doesn't mention how to handle that case. It would be nice for it to talk about that explicitly, especially since it is an exception to the "axe will pass messages which are round-trippable through JSON.stringify" rule.

Reply:
(if the intent is to pass it on as-is to `replyHandler`s as part of `ReplyData` `message`s, it would be good to update the type of the `message` property in `ReplyData` to match the `message` parameter in `Responder`)